src/app/components/client/toolbar/AnnouncementDialog.tsx

```diff
@@ -0,0 +1,404 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import Image from "next/image";
+import { useButton, useOverlayTrigger } from "react-aria";
+import { useL10n } from "../../../hooks/l10n";
+import { useRef, useState } from "react";
+import { useOverlayTriggerState } from "react-stately";
+import { Popover } from "../Popover";
+import { AnnouncementsIcon, CloseBtn } from "../../server/Icons";
+import styles from "./AnnouncementDialog.module.scss";
+import { UserAnnouncementWithDetails } from "../../../../../src/db/tables/user_announcements";
+import { LocalizedAnnouncementString } from "../../../(proper_react)/(redesign)/(authenticated)/admin/announcements/AnnouncementsAdmin";
```

Comment:
suggestion: Should we move `LocalizedAnnouncementString` out of the `/admin` directory if it is used for all users?